### ISSUE TITLE
Improve docs: `shuffle=true` means bootstrap sampling of training data for each epoch

### ISSUE THREAD

**Issue by dpaetzel** [xcsf-dev/xcsf#176]
I just went over the code tracing what `shuffle=true` in `XCS().fit(X, y, shuffle=true)` actually means in the context of the epochs defined by `PERF_TRIALS`. It appears that, if `shuffle=true`, the training data used in each epoch is actually a bootstrap sample from `(X, y)`? [1] These samples have a size of `MAX_TRIALS/PERF_TRIALS`. This seems reasonable but it took me a few minutes to find since I'm not too familiar with that part of the code and the indexing/partitioning information (which would be relevant if `shuffle=false`) is only discarded as late as `xcs_supervised_sample`. Maybe the documentation could be a bit more explicit with respect to that?

[1] I mainly wasn't sure whether the epoch samples form a partition or could overlap.

**Comment by rpreen** [comment]
I don't think it is equivalent to bootstrap sampling.

In each iteration, a single random training example is selected (with replacement) and a single update performed. This is what's called "online learning" or "stochastic training". It's similar to doing SGD with a batch size of 1, where training examples are sampled randomly with replacement. These samples are drawn `MAX_TRIALS` number of times in total, but for learning metrics the system performance is averaged over the most recent `PERF_TRIALS` number of iterations. 

Does that make sense? I can add this to the wiki (and you can feel free to tweak).

**Comment by dpaetzel** [comment]
OK, yes, at the level of the individual updates it's more akin to unbatched incremental updates, I agree. At the level of epochs (since the callbacks work at that level this is probably a useful term nevertheless) it's more bootstrap-like but I guess that unbatched updates should be the main mental model after all. Thank you!

**Comment by rpreen** [comment]
I've added some notes to the "hyperparameters" and "fitting" sections on the wiki to describe what is meant by a "trial" and how the sampling is done for fitting with `shuffle=True`. The term "epoch" has always been a bit awkward since it's an online system and it's only really for referring to when performance is displayed and callbacks run.

Feel free to make any adjustments to the wiki or add anything to make anything clearer. These discussions always make it better.
